File: Unit2-04/sam-stack.ts
/*
 * Class that defines string stack.
 *
 * @author  Samuel Webster
 * @version 1.0
 * @since   2024-01-01
 */

export class SamStack {
  private strstack: string[]

  // variables
  constructor() {
    self.strstack = []
  }

  // checks if stack is empty
  public get isEmpty() {
    return self.strstack.length == 0
  }

  // returns stack size
  public get size() {
    return self.strstack.length
  }

  // show full stack seperated by commas
  public get show() {
    let values = ""
    if (self.strstack[0]) {
      for (let counter = 0; counter < self.strstack.length; counter++) {
        values = values + self.strstack[counter] + ", "
      }
      values = values.substring(0, values.length-2)
    } else {
      values = "stack is empty"
    }
    return values
  }

  // push item to end of list
  public push(input) {
    self.strstack.push(input)
  }

  // pop item from end of list and return value
  public pop() {
    let removed = ""
    if (self.strstack[0]) {
      removed = self.strstack[self.strstack.length - 1]
    } else {
      removed = "nothing to remove"
    }
    self.strstack.splice(-1)
    return removed
  }

  // return top item of stack
  public get peek():string {
    return self.strstack[self.strstack.length - 1]
  }

  // clear all items off stack
  public clear() {
    self.strstack = []
  }
}

File: Unit2-04/Extra/SamStack.java
/*
* This program calculates the energy released when mass is converted to energy.
*
* @author  Samuel Webster
* @version 1.0
* @since   2024-03-25
*/

import java.util.ArrayList;
import java.util.List;

/**
* This is my stack.
*/
public class SamStack {
    /** The internal variable that stores stack. */
    private List<String> strstack = new ArrayList<String>();

    /**
    * Push a string to end of stack.
    *
    * @param input user input string
    */
    public void push(String input) {
        strstack.add(input);
    }

    /**
    * Get if stack is empty.
    *
    * @return get if stack is empty
    */
    public boolean isEmpty() {
        return strstack.size() == 0;
    }

    /**
    * Get top stack item.
    *
    * @return top stack item
    */
    public String peek() {
        String value = "";
        if (strstack.size() > 0) {
            value = strstack.get(strstack.size() - 1);
        } else {
            value = "stack is empty";
        }
        return value;
    }

    /**
    * Get stack size.
    *
    * @return get length
    */
    public int size() {
        return strstack.size();
    }

    /**
    * Show all items on stack.
    *
    * @return the full stack
    */
    public String show() {
        String values = "";
        if (strstack.size() > 0) {
            for (int counter = 0; counter < strstack.size(); counter++) {
                values = values + strstack.get(counter) + ", ";
            }
            if (strstack.size() > 0) {
                values = values.substring(0, values.length() - 2);
            }
        } else {
            values = "stack is empty";
        }
        return values;
    }

    /**
    * Clear all items on stack.
    */
    public void clear() {
        strstack.clear();
    }

    /**
    * Pop last item from stack.
    *
    * @return the popped item
    */
    public String pop() {
        String removed = "";
        if (strstack.size() > 0) {
            removed = strstack.get(strstack.size() - 1);
            strstack.remove(strstack.size() - 1);
        } else {
            removed = "nothing to remove";
        }
        return removed;
    }
}
